test/helpers.js

```diff
@@ -67,34 +110,87 @@ export const stripPositions = function(expr) {
     return expr;
 };
 
-export const parseAndSetResult = function(expr, result,
-                                          settings = new Settings()) {
+export const Mode = {
+    PARSE: {
+        apply: parseTree,
+        get: getParsed,
+        noun: 'parsing',
+        Verb: 'Parse',
+    },
+    BUILD: {
+        apply: _getBuilt,
+        get: getBuilt,
+        noun: 'building',
+        Verb: 'Build',
+    },
+};
+
+const getTree = (expr, settings, mode) => {
+    const result = expectKaTeX(expr, settings, mode);
     try {
-        return parseTree(expr, settings);
+        expect(result).toHavePassed();
     } catch (e) {
-        result.pass = false;
-        if (e instanceof ParseError) {
-            result.message = () =>
-                `'${expr}' failed parsing with error: ${e.message}`;
-        } else {
-            result.message = () =>
-                `'${expr}' failed parsing with unknown error: ${e.message}`;
-        }
+        Error.captureStackTrace(e, mode.get); // remove helpers in stack entries
+        throw e;
     }
+    return result._tree;
 };
 
-export const buildAndSetResult = function(expr, result,
-                                          settings = new Settings()) {
+export const expectKaTeX = (expr, settings = new Settings(), mode, isNot,
```

Comment:
Does the default `settings` work when `undefined` is passed as the second arg?

Reply:
I really like that we're supporting `.not` now, but this helper function has complicated the implementation quite a bit.  I'd rather have a few small helper functions that each of the `toParse()`, `toBuild()`, etc. call then a single function that tries to do it all.  Maybe we could add something format the message in addition to the function you added to clean up the stack trace.

Reply:
@kevinbarabash 
> Does the default settings work when undefined is passed as the second arg?

Yes. Default function parameters allow formal parameters to be initialized with default values if no value or `undefined` is passed.